### DISCUSSION TITLE
Automatically link all URLs everywhere

### DISCUSSION THREAD

**Discussion by bugeats** [standardnotes/forum#2529] [Suggestions]
Right now the experience of simply pasting in a URL for later reference is incomplete. Ideally, anywhere a URL is displayed, it would be click'able. This would be especially useful for plain text formats. Creating a named link in Markdown is fiddly and time consuming. There should be no need to name a link in order to click a simple URL. 

Thanks for your consideration.

**Comment by atmoio** [comment]
It's a much tricker problem than you'd expect, but totally agree this would be very useful.

**Comment by bugeats** [comment]
I don't doubt it! Hopefully no need to reinvent the URL detection bit. Something like [anchorme.js](https://alexcorvi.github.io/anchorme.js/) might work.

I can see how running this routine on all views and all editors would be tricky for sure. Personally, I'd be happy to have it in plain text and task editor modes.

**Comment by bschlagel** [comment]
I would also love this. One of the top missing features for me personally.

Here's another JS library for parsing links, seems to be used by many projects (I found via Discourse) and work quite well: https://github.com/markdown-it/linkify-it

Also noting there's a related issue open re: the minimal markdown editor: https://github.com/sn-extensions/minimal-markdown-editor/issues/14

But I for one would be very happy to have this in the plaintext editor, if easier to implement there!

**Comment by meixger** [comment]
Will this be considered / planned?

When using the **Markdown Alternative** editor the only way to open URLs is a right-click > "Open link in new tab|window"

I would prefer a `<shift> ` or `<ctrl>`  `click`

related [No way to navigate to links in Markdown Visual notes #1605](https://github.com/standardnotes/forum/issues/1605)
